torch_geometric/loader/link_neighbor_loader.py

```diff
@@ -79,16 +94,46 @@ def _create_label(self, edge_label_index, edge_label):
 
         edge_label = torch.cat([pos_edge_label, neg_edge_label], dim=0)
 
-        return edge_label_index, edge_label
+        return edge_label_index, edge_label, edge_label_time
+
+    def _modify_node_time(self, edge_label_index, edge_label_time):
+        """If `edge_label_time` is `None` return `self.node_time_dict`.
+        Else, for edges in a batch replace `src` and `dst`
+        node times by the min across all edge times."""
+        if edge_label_time is None:
+            return self.node_time_dict
+
+        def update_time(node_time_dict, index, input_type, num_nodes):
```

Comment:
I think this is overly complicated. I think
```python
scatter_min(edge_label_time, index, out=node_time_dict[input_type])
```
should do the tricky already.

Also: Don't we want to use max here? I thought the goal is to increase the timestamp so that we can sample neighbors that appeared after the seed node.

Reply:
Thanks that simplified things a lot.
If a node is a part of multiple links, then the nodes time should be update to the min of all those links, that way we'll only sample nodes before any of those links.

Reply:
Ok, that means we use the min for `edge_label_index`, but if we update `node_time_dict` we want to take the `max`, right?

Reply:
Yes your right. Updated that.